### ISSUE TITLE
Regression in 0.62.0: Style/TrailingCommaInArguments erroneously corrects heredoc

### ISSUE THREAD

**Issue by jaredbeck** [rubocop-hq/rubocop#6627]
## Expected behavior

Rubocop should not register any offenses for the following:

```ruby
order(
  <<-SQL.squish
    #{redacted}.a ASC,
    #{redacted}.b ASC,
    #{redacted}.c ASC
  SQL
)
```

## Actual behavior

Registers erroneous offense.

```plain
redacted.rb:3:22: C: Style/TrailingCommaInArguments: Avoid comma after the last parameter of a method call.
    #{redacted}.a ASC,
                     ^
```

`--auto-correct` erroneously removes all commas in the heredoc.

## Steps to reproduce the problem

Put the above in a file and run rubocop 0.62.0 (default config) on it.

## RuboCop version

```
bundle exec rubocop -V
0.62.0 (using Parser 2.5.3.0, running on ruby 2.5.3 x86_64-darwin18)
```

## Analysis

This is a regression between 0.61.1 and 0.62.0. Just looking at the changelog, it was probably introduced by https://github.com/rubocop-hq/rubocop/pull/6581 but I have not confirmed that yet.

**Comment by abrom** [comment]
It looks like the heredoc is no longer being detected as such; caused by https://github.com/rubocop-hq/rubocop/commit/27341536e6cbea0105a6e3ee887a2f292517538c#diff-0f2f566812ba3cfbc81aed3cd98391a4R80

Per the associated spec change, I believe what the change is trying to test is for end of line comments. Unfortunately it also hits on any string interpolation within heredoc.

Could change the `heredoc?` method definition to something like:

```
def heredoc?(source_after_last_item)
  source_after_last_item !~ /^\s*#[^{]/ && source_after_last_item =~ /\w/
end
```

But that seems somewhat brittle as it would trip over something like `foo #{ valid comment with trailing comma,`. May be the safest solution until something better comes along?

**Comment by fujimura** [comment]
Also, method call in interpolation:

```
foo(
  <<~BODY
    #{bar(1, 2)}
  BODY
)
```

```
C: Style/TrailingCommaInArguments: Avoid comma after the last parameter of a method call.
    #{bar(1, 2)}
```

**Comment by fujimura** [comment]
> Can you confirm the comma in question is the one between the 1 and the 2 ?

Yes, I had commented to report the problem with `bar(1, 2)` in heredoc as a relevant issue, but I've found another issue(#6629) which solely addressing the issue I've commented after my comment.
